src/bin/fixPluginIds.ts

```diff
@@ -0,0 +1,106 @@
+import { asArray, asMap, asObject, asString } from 'cleaners'
+import js from 'jsonfile'
+import nano from 'nano'
+
+import { asDbTx } from '../types'
+import { datelog } from '../util'
+
+const asApp = asObject({
+  _id: asString,
+  _rev: asString,
+  appId: asString,
+  appName: asString,
+  pluginIds: asMap(asMap(asString))
+})
+const asApps = asArray(asApp)
+
+const asPartitionResult = asObject({
+  rows: asArray(
+    asObject({ id: asString, value: asObject({ rev: asString }), doc: asDbTx })
+  )
+})
+
+const PLUGIN_ID_MAP = {
+  fox: 'foxExchange'
+}
+
+const config = js.readFileSync('./config.json')
+const nanoDb = nano(config.couchDbFullpath)
+const reportsApps = nanoDb.use('reports_apps')
+const reportsProgress = nanoDb.use('reports_progresscache')
+const reportsTransactions = nanoDb.use('reports_transactions')
+
+async function fixPluginIds(): Promise<void> {
+  try {
+    // get the contents of all reports_apps docs
+    const query = {
+      selector: {
+        appId: { $exists: true }
+      },
+      limit: 1000000
+    }
+    const rawApps = await reportsApps.find(query)
+    const apps = asApps(rawApps.docs)
+    for (const app of apps) {
+      for (const pluginId in PLUGIN_ID_MAP) {
+        const partition = `${app.appId}_${pluginId}`
+        const result = await reportsTransactions.partitionedList(partition, {
```

Comment:
This query should be paged to process no more than 100 transactions at a time and loop as necessary.